bokeh/core/has_props.py

```diff
@@ -605,10 +605,16 @@ def apply_theme(self, property_values):
         elif hasattr(self, '__themed_values__'):
             delattr(self, '__themed_values__')
 
+        # Property container values might be cached even if unmodified. Invalidate
+        # any cached values that are not modified at this point.
+        for k, v in old_values.items():
+            if isinstance(v, PropertyValueContainer) and k in self._unstable_default_values:
+                del self._unstable_default_values[k]
```

Comment:
this is deleting defaults that were not themed as well I think? though I guess that's harmless other than a minor efficiency cost. An effect of it for example would be to re-run the default function for defaults that have a default provided by a function. If the function returns a different value on the second run, it could get weird.